pandas/core/generic.py

```diff
@@ -2966,7 +2967,10 @@ def to_latex(
             The LaTeX caption to be placed inside ``\caption{}`` in the output.
 
             .. versionadded:: 1.0.0
-
+        short_caption : str, optional
```

Comment:
rather than another keyword parameter, maybe caption could be changed to accept Optional[Union[str, Tuple[str,str]]] to allow the short caption to be passed.

Reply:
I can do you this if you think this is the best way.
My concern is that in this case caption kwarg will be too long (another concern is that further unpacking into caption and short_caption from either string or tuple will look imperfect, but it is manageable).

Reply:
@simonjayhawkins, can you please confirm that this is the option to proceed with?

Reply:
i've added the api design and needs discussion labels. we should get some opinions from others before proceeding. Changing the api after the event is not so easy. cc @WillAyd @jreback  @toobaz

Reply:
I wouldn't bother about the caption kwarg being too long, while I do find that the additional argument _might_ be slightly easier to document. But all in all I have a slight preference for avoiding the new argument (and the new error message) and resorting to the tuple version.
I would actually phrase the docs as something like "a tuple (short_caption, full_caption), which will result in \caption[short_caption]{caption}; if a single string is passed, no short caption will be set".

Reply:
Should I wait for the decision from the remaining stakeholders?

Reply:
I don't have a strong preference. Looks like everyone so far has preferred the tuple version though so I'd say go ahead with that

There is some ambiguity as to whether "xy" is a single caption or two if you unpack. Maybe not a big deal but worth calling out

Reply:
I made the changes required, including testing corner cases like "xy" caption.